File: src/main/java/br/ufsc/bridge/platform/validation/rules/CepRule.java
package br.ufsc.bridge.platform.validation.rules;

import br.ufsc.bridge.platform.validation.engine.Rule;
import br.ufsc.bridge.platform.validation.util.Util;

public class CepRule implements Rule<String> {

	private MatchRule match = new MatchRule("^[0-9]{8}$");

	@Override public String validate(String value) {
		if (!Util.isEmpty(value) && !this.match.isValid(value)) {
			return "O CEP informado é inválido";
		}
		return null;
	}

}

File: src/main/kotlin/br/ufsc/bridge/platform/validation/rules/CepRule.kt
package br.ufsc.bridge.platform.validation.rules

import br.ufsc.bridge.platform.validation.engine.Rule
import br.ufsc.bridge.platform.validation.util.Util

class CepRule : Rule<String?> {
    private val match = MatchRule("^[0-9]{8}$")

    override fun validate(value: String?): String? {
        return if (!Util.isEmpty(value) && !this.match.isValid(value)) {
            "Campo é inválido"
        } else null
    }
}
